test/custom-field-keyboard-test.html

```diff
@@ -0,0 +1,80 @@
+<!doctype html>
+
+<head>
+  <meta charset="UTF-8">
+  <title>vaadin-custom-field tests</title>
+  <script src="../../web-component-tester/browser.js"></script>
+  <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+  <script src="../../iron-test-helpers/mock-interactions.js"></script>
+  <link rel="import" href="../../test-fixture/test-fixture.html">
+  <link rel="import" href="../vaadin-custom-field.html">
+  <link rel="import" href="helpers.html">
+</head>
+
+<body>
+  <test-fixture id="default">
+    <template>
+      <vaadin-custom-field>
+        <input type="text">
+        <input type="text">
+        <input type="number">
+        <input type="number">
+      </vaadin-custom-field>
+    </template>
+  </test-fixture>
+
+  <script>
+    describe('keyboard navigation', function() {
+      var customField;
+
+      function tab(target) {
+        MockInteractions.keyDownOn(target, 9, [], 'Tab');
+      }
+
+      function shiftTab(target) {
+        MockInteractions.keyDownOn(target, 9, 'shift', 'Tab');
+      }
+
+      beforeEach(function() {
+        customField = fixture('default');
+      });
+
+      it('should fire `internal-tab` event on Tab', () => {
+        const internalTabSpy = sinon.spy();
+        customField.addEventListener('internal-tab', internalTabSpy);
+        for (let i = 0; i < 3; i ++) {
+          tab(customField.inputs[i]);
+        }
+
+        expect(internalTabSpy.callCount).to.equal(3);
+      });
+
+      it('should not fire `internal-tab` event on Tab from the last input', () => {
+        const internalTabSpy = sinon.spy();
+        customField.addEventListener('internal-tab', internalTabSpy);
+        customField.inputs[4];
```

Comment:
This wouldn't fire an event in any case. same below

Reply:
Fixed.